tests/conftest.py

```diff
@@ -74,6 +74,18 @@ def backend(backend_name):
     yield get_backend(backend_name)
 
 
+@pytest.fixture(autouse=True)
+def clear():
+    yield
+
+    from qibo.transpiler.pipeline import Passes
+
+    _Global._backend = None
+    _Global._transpiler = None
```

Comment:
Yes, I knew the current one was not going to work. We had to extend it.

In any case, this is a fully internal usage, and it's an enhancement for the user (since he can always change the backend with `set_backend()` to a specific one, or reload the environment).

I split as an issue on its own, you could resolve the conversation whenever you wish.